### DISCUSSION TITLE
spatial_domain = dde.geometry.Polygon and self.normal = self.normal / np.linalg.norm(self.normal, axis=1).reshape(-1, 1)

### DISCUSSION THREAD

**Discussion by Chaztikov** [lululxvi/deepxde#704] [Q&A]
Hi, I'm wondering if there is a quick solution to the following problem: 
(I apologize for my brevity, I hope I don't sound rude)
  
spatial_domain = dde.geometry.Polygon(pts_boundary)
yields
/home/chaztikov/.local/lib/python3.6/site-packages/deepxde/geometry/geometry_2d.py:353: RuntimeWarning: invalid value encountered in true_divide
self.normal = self.normal / np.linalg.norm(self.normal, axis=1).reshape(-1, 1)

What is the cause of this issue?  Too many points? Repeated points? Corners?

If I remove the corners, is that sufficient? If I randomly sample too few points, and retry with

        nboundarypts = int(pts_boundary.shape[0] * 0.01 )
        print(nboundarypts)
        iboundarypts = np.random.choice(pts_boundary.shape[0], nboundarypts)                
        spatial_domain = dde.geometry.Polygon(pts_boundary[iboundarypts])

Then

        pts = spatial_domain.uniform_boundary_points( nboundarypts )
        plt.scatter(pts.T[0],pts.T[1]);plt.show()
Yields something like the first figure (the second figure is the original dataset)

![image](https://user-images.githubusercontent.com/9398385/170534653-0187666e-fb1b-4d47-a1bb-b882b06d0769.png)

![image](https://user-images.githubusercontent.com/9398385/170532060-1b70c064-db5c-499f-b5b8-d5c012d29af8.png)

**Comment by lululxvi** [comment]
What is the value of `pts_boundary`?
